chapter05/red/exercise2.pl

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+use List::Util qw( reduce );
+
+my $fn = shift @ARGV // 'coconet.dat';
+open my $fh, '<', $fn;
+
+my %stats;
+while( <$fh> ) {
+  next if /^\s*(#.*)?$/;
```

Comment:
Have to agree with @Utopia1138 here, as much as it pains me.  Except, if anything, braceless single statement conditional bodies would be the worst choice of them all!